### ISSUE TITLE
Improve getting started for "both" case

### ISSUE THREAD

**Issue by bajamircea** [aaugustin/websockets#143]
Hi,

To get rid of the message `(This code looks convoluted. If you know a more straightforward solution, please let me know about it!)` in the Getting started guide https://websockets.readthedocs.io/en/stable/intro.html#both

I would suggest replacing
```
async def handler(websocket, path):
    while True:
        listener_task = asyncio.ensure_future(websocket.recv())
        producer_task = asyncio.ensure_future(producer())
        done, pending = await asyncio.wait(
            [listener_task, producer_task],
            return_when=asyncio.FIRST_COMPLETED)

        if listener_task in done:
            message = listener_task.result()
            await consumer(message)
        else:
            listener_task.cancel()

        if producer_task in done:
            message = producer_task.result()
            await websocket.send(message)
        else:
            producer_task.cancel()
```

with something on the lines of

```
async def listener_handler(websocket):
    while True:
        message = await websocket.recv()
        await consumer(message)

async def producer_handler(websocket):
    while True:
        message = await producer()
        await websocket.send(message)

async def handler(websocket, path):
    listener_task = asyncio.ensure_future(listener_handler(websocket))
    producer_task = asyncio.ensure_future(producer_handler(websocket))
    done, pending = await asyncio.wait(
        [listener_task, producer_task],
        return_when=asyncio.FIRST_COMPLETED)

    for t in pending:
        t.cancel()
```

Do you want me to elaborate why?

Cheers,
Mircea

**Comment by aaugustin** [comment]
Yes that seems a bit more efficient, even though it takes the same number of lines and has the same complexity.

**Comment by bajamircea** [comment]
Do you want me to create pull request with a commit?

**Comment by aaugustin** [comment]
Yes please.

Can you replace the `t` variable with `task` in your version and remove "(This code looks convoluted...)" sentence?

Thanks!

**Comment by aaugustin** [comment]
Related PR merged in c25954538aab85c259dc37e8235c1362d9543146.
